tools/taskcluster/decision_task.py

```diff
@@ -0,0 +1,70 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import json
+import os
+
+import taskcluster
+
+from tasks import schedule_task_graph, TaskBuilder
+
+
+def pr(builder, branch):
+    return (taskcluster.slugId(), builder.craft_pr_task(branch)),
+
+
+def master(builder):
+    return (taskcluster.slugId(), builder.craft_master_task()),
+
+
+def release(builder, tag):
+    return (taskcluster.slugId(), builder.craft_release_task(tag)),
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest='command')
+
+    pr_parser = subparsers.add_parser('pull-request')
+    pr_parser.add_argument('author')
+    pr_parser.add_argument('branch')
+
+    master_parser = subparsers.add_parser('master')
+    master_parser.add_argument('author')
+
+    release_parser = subparsers.add_parser('release')
+    release_parser.add_argument('tag')
+
+    result = parser.parse_args()
+    command = result.command
+    task_group_id = os.environ['TASK_ID']
+    repo_url = os.environ['MOBILE_HEAD_REPOSITORY']
+    commit = os.environ['MOBILE_HEAD_REV']
+    if command == 'pull-request':
+        builder = TaskBuilder(result.author, repo_url, commit, task_group_id)
+        ordered_tasks = pr(builder, result.branch)
+    elif command == 'master':
+        builder = TaskBuilder(result.author, repo_url, commit, task_group_id)
+        ordered_tasks = master(builder)
+    elif command == 'release':
+        builder = TaskBuilder('firefox-tv@mozilla.com', repo_url, commit, task_group_id)
+        ordered_tasks = release(builder, result.tag)
+    else:
+        raise ValueError('A command ("pull-request", "master" or "release") must be provided to '
+                         'the decision task')
+
+    full_task_graph = schedule_task_graph(ordered_tasks)
+    with open('task-graph.json', 'w') as f:
```

Comment:
Nit: Why don't we use 
```py
    populate_chain_of_trust_task_graph(full_task_graph)
    populate_chain_of_trust_required_but_unused_files()
```

anymore? Their names provide more context about what's going on. Not a blocker

Reply:
Right! I kept a comment for the "unused files", but I didn't leave a comment for `task-graph.json`, good eye.
I didn't write the functions because I wasn't sure that they added enough non-documentation-y value :smile: